### PR TITLE
POST the health study consent upon registering by a UK citizen.

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

[comment]: <> (Description of the feature or fix)
[comment]: <> (Include link to Figma designs, if relevant)
[comment]: <> (Include link to Notion ticket, if relevant)

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

## Screenshots

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Checklist (for merger)

- [ ] Make sure the right merge strategy is chosen, commit merge into master and squash merge into develop.

### CODE REVIEW

**Review by bionara — APPROVED** [review]

File: src/core/consent/ConsentService.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
+import appConfig from '@covid/appConfig';
 import { ApiClientBase } from '@covid/core/api/ApiClientBase';
 import { AsyncStorageService } from '@covid/core/AsyncStorageService';
 import { TConsent } from '@covid/core/user/dto/UserAPIContracts';
 
 type TConsentDocument = 'US' | 'UK' | 'SE' | '' | 'US Nurses' | 'Health Study Consent';
 
 interface IConsentService {
-  postConsent(document: TConsentDocument, version: string, privacy_policy_version: string): void; // TODO: define return object
+  postHealthStudyConsent(): void;
```
Comment (andyguest) [inline src/core/consent/ConsentService.ts:9]: You've made this interface less generic so I don't really like the change.  Is there really no one using this elsewhere?  Why don't US and SE people sign consents too?

**Review by andyguest — COMMENTED** [review]
Big question on this.  I absolutely must be missing some automatic consent stuff happening in the backend.

File: src/core/consent/ConsentService.ts (hunk 1 above)
Comment (andyguest) [reply src/core/consent/ConsentService.ts:9]: OK, I've filled in all the gaps in my understanding now.  New proposal: don't make this fix in the front end after all.

You may want to add a "for historical interest" comment to ReconsentRequestConsentScreen explaining that the postConsent() is not required for *new* registrants from the UK who will have this Consent provisioned automatically in the backend as of my upcoming PR.

Comment (aikewoody) [reply src/core/consent/ConsentService.ts:9]: @andyguest So this is gonna be taken care of by the backend when people from the UK register?

Comment (andyguest) [reply src/core/consent/ConsentService.ts:9]: This is very much still up for discussion at 10:45 :-)  The current tight-coupling doesn't feel correct and I wrote this comment before the slack reconsideration.

File: src/core/consent/ConsentService.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export class ConsentService extends ApiClientBase implements IConsentService {
     version: '',
   };
 
-  public async postConsent(document: TConsentDocument, version: string, privacy_policy_version: string) {
+  public async postConsent(document: TConsentDocument, privacy_policy_version: string, version: string) {
```
Comment (andyguest) [inline src/core/consent/ConsentService.ts:22]: I'm not sure reordering these parameters is a good idea.  The 'version' parameter refers directly to the document so they should be next to each other in the list, whereas privacy_policy_version is separate (and ought to be properly separated in the future!)

**Review by andyguest — APPROVED** [review]
Approval with one disclaimer, which you can choose to address or not.

**Review by andyguest — APPROVED** [review]
:-) 👍

File: src/features/register/ConsentScreen/ConsentScreen.tsx [hunk 3]
```diff
@@ -31,6 +31,11 @@ export default React.memo((props: TProps) => {
     }
     if (isGBCountry()) {
       await consentService.setConsentSigned('UK', appConfig.consentVersionUK, appConfig.privacyPolicyVersionUK);
+      await consentService.postConsent(
+        'Health Study Consent',
```
Comment (bionara) [inline src/features/register/ConsentScreen/ConsentScreen.tsx:35]: better as a const

**Review by bionara — APPROVED** [review]

**Pull request merged**